### PR TITLE
Add ECMWF cfgrib: open GRIB files with xarray

### PR DESCRIPTION
**Pull request by alexamici**
Hi, this recipe adds the ability to work with the very popular GRIB file format in `xarray` on Linux, MacOS and Windows (with the next release).

The need for a conda package is due to the `eccodes` dependency.

### CODE REVIEW

**Comment by alexamici** [comment]
@conda-forge/core this PR should be ready and safe to merge. Failures are the known issue #7715.

File: recipes/cfgrib/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+test:
+  imports:
+    - cf2cdm
+    - cfgrib
+  commands:
+    - cfgrib --help
+
+about:
+  home: https://github.com/ecmwf/cfgrib
+  license: Apache Software
```
Comment (chrisburr) [inline recipes/cfgrib/meta.yaml]:
```suggestion
  license: Apache-2.0
```

File: recipes/cfgrib/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  license_family: APACHE
```
Comment (chrisburr) [inline recipes/cfgrib/meta.yaml]:
```suggestion
  license_family: Apache
```

File: recipes/cfgrib/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  license_file: LICENSE
+  summary: Maps GRIB files to the NetCDF Common Data Model with CF Convention using ecCodes
+  doc_url: https://github.com/ecmwf/cfgrib
+  dev_url: https://github.com/ecmwf/cfgrib
+
+extra:
+  recipe-maintainers:
+    - alexamici
+    - StephanSiemen
```
Comment (chrisburr) [inline recipes/cfgrib/meta.yaml]: @StephanSiemen Are you happy to be listed as maintainer of this recipe?

File: recipes/cfgrib/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+build:
+  noarch: python
+  number: 0
+  entry_points:
+    - cfgrib=cfgrib.__main__:cfgrib_cli
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
+
+requirements:
+  host:
+    - attrs
```
[the remark this answers is not in the document]
Comment (alexamici) [reply recipes/cfgrib/meta.yaml]: I'm new to conda and honesty I don't understand what `requirements - host` is. I'll trust you on this.

File: recipes/cfgrib/meta.yaml (hunk 3 above)
Comment (alexamici) [reply recipes/cfgrib/meta.yaml]: I must say that I didn't explicitly ask him 😆, but he is the head of the development section of ECMWF that commissioned to my company the development of *cfgrib* and we usually co-maintain all distribution channels.

File: recipes/cfgrib/meta.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (chrisburr) [reply recipes/cfgrib/meta.yaml]: There  is a full explanation [here](https://conda.io/projects/conda-build/en/latest/source/resources/define-metadata.html#host) but for a `pip install`ed recipe the `host` requirements are `python`, `pip` and anything that is explicitly used by the `setup.py` file.

File: recipes/cfgrib/meta.yaml (hunk 3 above)
Comment (chrisburr) [reply recipes/cfgrib/meta.yaml]: It's standard for us to check that any additional maintainers are happy before merging new recipes (most of the time it's a formality, but occasionally people say no).

If you'd like it be merged faster you can remove this line and re-add him in the feedstock (which you can edit directly).

Comment (StephanSiemen) [reply recipes/cfgrib/meta.yaml]: Hi, I am very happy to be added as a maintainer. Thanks @alexamici for the work!

**Pull request merged**